src/Functions/Public/catalog-service/Get-vRAResource.ps1

```diff
@@ -297,4 +237,39 @@
 
     }
 
-}
\ No newline at end of file
+}
+
+Function New-vRAObjectResource {
```

Comment:
At the moment, our build script will export this as a public function. Could you modify the name to be:

```
function intNewvRAObjectResource
```

See here for reference: https://github.com/jakkulabs/PowervRA/blob/3a0b749c6d0bf90224c6e3dd51c83a681cb529a2/src/Functions/Public/catalog-service/Get-vRAResourceActionRequestTemplate.ps1#L53

Also we tend to use [PSCustomObject] for building objects. Could you change line 249 - 274 to:

```
[PSCustomObject]@{
        ResourceId = $Data.ResourceId
        BusinessGroupId = $Data.businessGroupId
        BusinessGroupName = $Data.data.MachineGroupName
        TenantId = $Data.tenantId
        CatalogItemLabel = $Data.data.Component
        ParentResourceId = $Data.parentResourceId
        HasChildren = $Data.hasChildren
        Data = $Data.data
        ResourceType = $Data.resourceType
        Name = $Data.name
        Description = $Data.description
        Status = $Data.status
        RequestId = $Data.requestId      
        Owners = $Data.owners
        DateCreated = $Data.dateCreated
        LastUpdated = $Data.lastUpdated
        Lease = $Data.lease
        Costs = $Data.costs
        CostToDate = $Data.costToDate
        TotalCost = $Data.totalCost
        Links = $Data.links
        IconId = $Data.iconId
}
```

See this function for reference:

https://github.com/jakkulabs/PowervRA/blob/3a0b749c6d0bf90224c6e3dd51c83a681cb529a2/src/Functions/Public/identity/Get-vRAVersion.ps1

Reply:
Ok for renaming the function of course.
About your remark about typing the object, the New-Object -TypeName PSObject cmdlet already type the object as PSCustomObject.
![image](https://user-images.githubusercontent.com/12493050/35635284-2c9e0e56-06ae-11e8-8893-a57df5267a0a.png)
And as I use splatting to dynamically set object properties, I think it's not necessary.
What's your point?

Reply:
@BlackCatDeployment It's more performant to use the [pscustomobject] way to create objects, rather than using New-Object. (See https://www.jonathanmedd.net/2011/09/powershell-v3-creating-objects-with-pscustomobject-its-fast.html for a rough comparison of speed difference) Also, it would be more consistent with how we have created objects in the rest of the project code.

Reply:
Ok done :)